GlottisFactory.h

```diff
@@ -0,0 +1,87 @@
+// ****************************************************************************
+// This file is part of VocalTractLab.
+// Copyright (C) 2020, Peter Birkholz, Dresden, Germany
+// www.vocaltractlab.de
+// author: Simon Stone
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+// ****************************************************************************
+
+#ifndef GLOTTISFACTORY_H
+#define GLOTTISFACTORY_H
+
+#include <map>
+#include <string>
+
+#include "Glottis.h"
+
+/// @brief Types of available Glottis objects
+enum GlottisModel
+{
+	GEOMETRIC_GLOTTIS,
+	TWO_MASS_MODEL,
+	TRIANGULAR_GLOTTIS,
+	NUM_GLOTTIS_MODELS
```

Comment:
That is one of Peter's conventions. He includes one last value in all his enums. It allows iterating over the "actual" enum values like so:
```
for (GlottisModel g = GEOMETRIC_GLOTTS; g < NUM_GLOTTIS_MODELS; ++g)
{
  // Do something
}
```
(BTW: This is only possible because "classic" `enum`s are just fancy `int`s. The more modern ``enum class`` does not allow this anymore.)

Reply:
I dont't like this, but I get it.